### PR TITLE
re-record Confidential Ledger tests after proxy fixes

### PR DESCRIPTION
**Pull request by christothes**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by JoshLove-msft** [comment]
Do we need to bump the Test Proxy version? https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L275

I'm curious how you were able to re-record without doing this.

**Comment by christothes** [comment]
> Do we need to bump the Test Proxy version? https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L275
> 
> I'm curious how you were able to re-record without doing this.

Yes - I was just waiting for that to publish since [this](https://github.com/Azure/azure-sdk-tools/pull/3699) merged.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 1]
```diff
@@ -330,6 +330,19 @@ public void TearDownRecordedTestClass()
...
+            if (Mode == RecordedTestMode.Record && options != null)
+            {
+                await _proxy.Client.SetRecordingTransportOptionsAsync(Recording.RecordingId, options).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// If set to true, the proxy will be configured to connect on ports 5000 and 5001. This is useful when running the proxy locally for debugging recorded test issues.
+        /// </summary>
+        protected bool UseLocalDebugProxy { get; set; }
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:344]: We should use the same pattern as SaveDebugRecordingsOnFailure property where we throw if this is running in CI. Also, group with the other properties, ideally right below SaveDebugRecordingsOnFailure.

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 2]
```diff
@@ -330,6 +331,29 @@ public void TearDownRecordedTestClass()
...
+                await _proxy.Client.SetRecordingTransportOptionsAsync(Recording.RecordingId, options).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// If set to true, the proxy will be configured to connect on ports 5000 and 5001. This is useful when running the proxy locally for debugging recorded test issues.
+        /// </summary>
+        protected bool UseLocalDebugProxy
+        {
+            get => _useLocalDebugProxy; set
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:347]:
```suggestion
            get => _useLocalDebugProxy; 
            set
```

File: sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs [hunk 3]
```diff
@@ -330,6 +331,29 @@ public void TearDownRecordedTestClass()
...
+            if (Mode == RecordedTestMode.Record && options != null)
+            {
+                await _proxy.Client.SetRecordingTransportOptionsAsync(Recording.RecordingId, options).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// If set to true, the proxy will be configured to connect on ports 5000 and 5001. This is useful when running the proxy locally for debugging recorded test issues.
+        /// </summary>
+        protected bool UseLocalDebugProxy
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs:345]: Can we move this to be with the other properties?

File: sdk/core/Azure.Core.TestFramework/src/TestProxy.cs [hunk 4]
```diff
@@ -134,7 +141,12 @@ private TestProxy(string proxyPath)
                 });
         }
 
-        public static TestProxy Start()
+        /// <summary>
+        /// Starts the test proxy
+        /// </summary>
+        /// <param name="debugMode">If true, the proxy will be configured to look for port 5000 and 5001, which is the default used when running the proxy locally in debug mode.</param>
+        /// <returns></returns>
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core.TestFramework/src/TestProxy.cs:148]:
```suggestion
        /// <returns>The started TestProxy instance.</returns>
```

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**